rfc/cart-api-v1.md

```diff
@@ -0,0 +1,84 @@
+# Version 1
+
+| Issue | Status |
+| ----- | ------ |
+| #88   | Draft  |
+
+## Data Models
+
+### `Cart`
+| Property | Type | Description |
+| --- | --- | --- |
+| `id` | UUID-like `str` | Cart ID, auto-generated by the web API |
+| `user_id` | UUID-like `str` | The ID of the user that creates the collection |
+| `cart_name` | str | Cart Name
+| `entity_type` | str | Entity Type |
+
+### `CartItem`
+| Property | Type | Description |
+| --- | --- | --- |
+| `id` | UUID-like `str` | Cart Item ID, auto-generated by the web API |
+| `cart_id` | UUID-like `str` | Associated `Cart.id` |
+| `entity_id` | UUID-like `str` | Entity ID |
+| `bundle_id` | UUID-like `str` | Bundle ID |
+| `bundle_version` | `str` | Bundle Version |
+
+## Web API Specification
+
+Please note that:
+1. All endpoints are protected by **a bearer token** ([RFC 6750](https://tools.ietf.org/html/rfc6750)).
+
+> For the initial release,
+> * the bearer token can be anything that we use to identify a user *until the web service integrates with the auth service (`https://auth.{DEPLOYMENT_STAGE}.data.humancellatlas.org`, [docs](https://allspark.dev.data.humancellatlas.org/dcp-ops/docs/wikis/Security/Authentication%20and%20Authorization/Setting%20up%20DCP%20Auth))*, and
+> * the client will generate a version 4 UUID string as dubbed as session ID (and the backend will treat the session ID as user ID) and the base64-encoded string of that UUID will be a bearer token.
+
+2. All CRUD endpoints take JSON request body on `POST` and `PUT`/`PATCH` requests. `id` is not required as either the ID will be auto-generated by the endpoint on `POST` request, or the ID is given on the URL and immutable on `PUT`/`PATCH` request.
+  * Generally, the response on `GET` request will return a single JSON object of the corresponding model if the ID is specified or the list of JSON object otherwise. *There will be some exceptions.*
+
+#### CRUD APIs for `Cart`
+
+| Method | Path | Description |
+| --- | --- | --- |
+| `POST` | `/resources/carts/` | Create a new cart |
+| `GET` | `/resources/carts/` | Retrieve a list of carts |
+| `GET` | `/resources/carts/{id}` | Retrieve a single cart (see more detail below) |
+| `PUT`/`PATCH` | `/resources/carts/{id}` | Either overwrite the whole cart or partially update the cart |
+| `DELETE` | `/resources/carts/{id}` | Delete a single cart (see more detail below) |
+
+#### `GET /resources/carts/{id}`
+
+Instead of returning `Cart` information, the response will be also returning the list of associated `CartItem`, as shown below:
```

Comment:
/carts/{id} should only return information about the cart itself.

/carts/{id}/items should be used for cart items.